### PR TITLE
Remove .script! hack

### PR DESCRIPTION
**Pull request by bkeepers**
This replaces #1786, which removes the `.script!` hack that was used for extension-less executable files. We now detect shebang before looking at the filename, so this is not necessary.

/cc @arfon @pchaigno

### CODE REVIEW

File: lib/linguist/samples.rb [hunk 1]
```diff
@@ -50,16 +50,13 @@ def self.each(&block)
             end
           else
             path = File.join(dirname, filename)
-
-            if File.extname(filename) == ""
-              raise "#{path} is missing an extension, maybe it belongs in filenames/ subdir"
```
Comment (pchaigno) [inline lib/linguist/samples.rb]: Does it mean that `filenames/` subdirectories are deprecated now?

Comment (bkeepers) [inline lib/linguist/samples.rb]: Nope, it now allows files without an extension to exist _outside_ of `filenames`.

Comment (pchaigno) [inline lib/linguist/samples.rb]: Okay, I see :-)

**Comment by pchaigno** [comment]
:+1:

**Pull request merged**